### PR TITLE
tl 1.0.0

### PR DESCRIPTION
**Pull request by ericLemanissier**
Specify library name and version:  **tl/1.0.0**

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by uilianries** [comment]
@ericLemanissier that recipe is already available on Conan Center:

https://github.com/conan-io/conan-center-index/blob/43e7daf87454a7f5b9473770b8ece6a388e1a978/recipes/tl-expected/all/conanfile.py

https://conan.io/center/tl-expected/20190710/?user=_&channel=_

**Comment by ericLemanissier** [comment]
this is not the same library: This pr is for https://github.com/TartanLlama/tl whereas the one you link is for https://github.com/TartanLlama/expected

File: recipes/tl/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+from conans import ConanFile, CMake, tools
+from fnmatch import fnmatch
```
Comment (uilianries) [inline recipes/tl/all/conanfile.py:2]:
```suggestion
```

File: recipes/tl/all/test_package/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+cmake_minimum_required(VERSION 2.8.12)
+project(PackageTest CXX)
+set(CMAKE_CXX_STANDARD 14)
```
Comment (uilianries) [inline recipes/tl/all/test_package/CMakeLists.txt:3]:
```suggestion
```

File: recipes/tl/all/test_package/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+add_executable(example example.cpp)
+target_link_libraries(example ${CONAN_LIBS})
```
Comment (uilianries) [inline recipes/tl/all/test_package/CMakeLists.txt:9]:
```suggestion
target_link_libraries(example ${CONAN_LIBS})
set_property(TARGET example PROPERTY CXX_STANDARD 14)
```

File: recipes/tl/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+class TlConan(ConanFile):
+    name = "tl"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://tl.tartanllama.xyz"
+    description = "tl is a collection of generic C++ libraries"
+    topics = ("conan", "c++", "utilities")
+    license = "CC0-1.0"
+    no_copy_source = True
+    _source_subfolder = "tl"
+
```
Comment (uilianries) [inline recipes/tl/all/conanfile.py:15]:
```suggestion
    def configure(self):
        minimal_cpp_standard = "14"
        if self.settings.compiler.cppstd:
            tools.check_min_cppstd(self, minimal_cpp_standard)

        minimal_version = {
            "gcc": "6.4",
            "clang": "5",
            "apple-clang": "10",
            "Visual Studio": "15"
        }

        compiler = str(self.settings.compiler)
        if compiler not in minimal_version:
            self.output.warn(
                "%s recipe lacks information about the %s compiler standard version support" % (self.name, compiler))
            self.output.warn(
                "%s requires a compiler that supports at least C++%s" % (self.name, minimal_cpp_standard))
            return

        version = tools.Version(self.settings.compiler.version)
        if version < minimal_version[compiler]:
            raise ConanInvalidConfiguration(
                "%s requires a compiler that supports at least C++%s" % (self.name, minimal_cpp_standard))
```

File: recipes/tl/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
+from conans import ConanFile, CMake, tools
```
Comment (uilianries) [inline recipes/tl/all/conanfile.py:1]:
```suggestion
from conans import ConanFile, CMake, tools
from conans.errors import ConanInvalidConfiguration
```

File: recipes/tl/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+class TlConan(ConanFile):
+    name = "tl"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://tl.tartanllama.xyz"
+    description = "tl is a collection of generic C++ libraries"
+    topics = ("conan", "c++", "utilities")
+    settings = "compiler"
+    license = "CC0-1.0"
+    no_copy_source = True
+    _source_subfolder = "tl"
```
Comment (uilianries) [inline recipes/tl/all/conanfile.py:15]:
```suggestion
    _source_subfolder = "tl"
    
```

Add new empty line, please

File: recipes/tl/all/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+            self.output.warn(
+                "%s recipe lacks information about the %s compiler standard version support" % (self.name, compiler))
+            self.output.warn(
+                "%s requires a compiler that supports at least C++%s" % (self.name, minimal_cpp_standard))
+            return
+
+        version = tools.Version(self.settings.compiler.version)
+        if version < minimal_version[compiler]:
+            raise ConanInvalidConfiguration(
+                "%s requires a compiler that supports at least C++%s" % (self.name, minimal_cpp_standard))
```
Comment (uilianries) [inline recipes/tl/all/conanfile.py:39]:
```suggestion
                "%s requires a compiler that supports at least C++%s" % (self.name, minimal_cpp_standard))

```

new empty line

**Comment by jgsogo** [comment]
From the documentation I see that `expected` is [one _component_ of the `tl` libraries](https://tl.tartanllama.xyz/en/latest/#getting-the-code) although they are separated into different repositories. They are `tl-expected`, `tl-optional`, `tl-function_ref` and `tl` (misc utilities)

We need to think twice about this (and also about llvm, boost, qt,...), is this a _framework_? should we package all these libraries into a single `tl` package and wait for _components feature_  to consume them separately? 

I really don't know what to do with this kind of libraries 😞

**Comment by ericLemanissier** [comment]
this case is a lot simpler that llvm, qt or boost though. These lib share the same namespace, but they are completely independent (none of these lib has any requirement).

**Pull request merged**